File: algo/src/test/java/com/hartwig/actin/algo/evaluation/tumor/HasBoneMetastasesTest.java
package com.hartwig.actin.algo.evaluation.tumor;

import static com.hartwig.actin.algo.evaluation.EvaluationAssert.assertEvaluation;
import com.hartwig.actin.algo.datamodel.Evaluation;
import com.hartwig.actin.algo.datamodel.EvaluationResult;
import org.junit.Test;

public class HasBoneMetastasesTest {

    private final HasBoneMetastases function = new HasBoneMetastases();

    @Test
    public void shouldBeUndeterminedWhenHasBoneLesionsIsNull() {
        Evaluation undetermined = function.evaluate(TumorTestFactory.withBoneLesions(null));
        assertEvaluation(EvaluationResult.UNDETERMINED, undetermined);
    }

    @Test
    public void shouldPassWhenHasBoneLesionsIsTrue() {
        Evaluation pass = function.evaluate(TumorTestFactory.withBoneLesions(true));
        assertEvaluation(EvaluationResult.PASS, pass);
    }

    @Test
    public void shouldFailWhenHasBoneLesionsIsFalse() {
        Evaluation fail = function.evaluate(TumorTestFactory.withBoneLesions(false));
        assertEvaluation(EvaluationResult.FAIL, fail);
    }
}

File: algo/src/test/kotlin/com/hartwig/actin/algo/evaluation/tumor/HasBoneMetastasesTest.kt
package com.hartwig.actin.algo.evaluation.tumor

import com.hartwig.actin.algo.evaluation.EvaluationAssert.assertEvaluation
import com.hartwig.actin.datamodel.algo.EvaluationResult
import org.junit.jupiter.api.Test

class HasBoneMetastasesTest {
    
    private val function: HasBoneMetastases = HasBoneMetastases()

    @Test
    fun shouldBeUndeterminedWhenHasBoneLesionsIsNull() {
        val undetermined = function.evaluate(TumorTestFactory.withBoneLesions(null))
        assertEvaluation(
            EvaluationResult.UNDETERMINED,
            undetermined,
            "Undetermined if bone metastases based on provided lesions"
        )
    }

    @Test
    fun shouldPassWhenHasBoneLesionsIsTrue() {
        val pass = function.evaluate(TumorTestFactory.withBoneLesions(true))
        assertEvaluation(EvaluationResult.PASS, pass, "Bone metastases in provided lesions")
    }

    @Test
    fun shouldFailWhenHasBoneLesionsIsFalse() {
        val fail = function.evaluate(TumorTestFactory.withBoneLesions(false))
        assertEvaluation(EvaluationResult.FAIL, fail, "No bone metastases in provided lesions")
    }
}